### PR TITLE
NEW: Save invoice payment with a AJAX request so the user don't send many request at the same time

### PR DESCRIPTION
**Pull request by kkhelifa-opendsi**
NEW: Save invoice payment with a AJAX request so the user don't send many request at the same time (when the user refresh the page when the action take too many time for some reason, ...)

### CODE REVIEW

**Comment by eldy** [comment]
Need a security with restrictedArea()

File: htdocs/compta/ajax/addpayment.php [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+// Load object
+if ($facid > 0) {
+	$ret = $object->fetch($facid);
+}
+
+$usercanissuepayment = !empty($user->rights->facture->paiement);
+
+$fieldid = 'rowid';
+$isdraft = (($object->statut == Facture::STATUS_DRAFT) ? 1 : 0);
+$result = restrictedArea($user, 'facture', $object->id, '', '', 'fk_soc', $fieldid, $isdraft, 1);
```
Comment (kkhelifa-opendsi) [inline htdocs/compta/ajax/addpayment.php:72]: restrictedArea() is already here so where should I add it ?

**Comment by kkhelifa-opendsi** [comment]
Can you merge it, please ?

**Comment by thomas-Ngr** [comment]
@eldy, could you have a new review or merge if everything is right ?

**Comment by lvessiller-opendsi** [comment]
@eldy 
Could you check and merge it ?

File: htdocs/compta/paiement.php [hunk 2]
```diff
@@ -459,6 +459,49 @@ function callForResult(imgId)
 						});';
 			print '	});'."\n";
 
+			//Add js for AutoFill
+			if ($action == 'add_paiement') {
```
Comment (eldy) [inline htdocs/compta/paiement.php:463]: Need to extend test with
&& !empty($conf->use_javascript_ajax)

**Comment by eldy** [comment]
Replaced with #26041

**Pull request closed without merging**